account_statement_import_sheet_file/tests/test_account_statement_import_sheet_file.py

```diff
@@ -0,0 +1,492 @@
+        cls.currency_usd.active = True
+        # Activate EUR for unit test, by default is not active
+        cls.currency_eur.active = True
+        cls.sample_statement_map = cls.env.ref(
+            "account_statement_import_sheet_file.sample_statement_map"
+        )
+        cls.AccountJournal = cls.env["account.journal"]
+        cls.AccountBankStatement = cls.env["account.bank.statement"]
+        cls.AccountStatementImport = cls.env["account.statement.import"]
+        cls.AccountStatementImportSheetMapping = cls.env[
+            "account.statement.import.sheet.mapping"
+        ]
+        cls.AccountStatementImportWizard = cls.env["account.statement.import"]
+        cls.suspense_account = cls.env["account.account"].create(
+            {
+                "code": "987654",
+                "name": "Suspense Account",
+                "account_type": "asset_current",
+            }
+        )
+        cls.parser = cls.env["account.statement.import.sheet.parser"]
+        # Mock the mapping object to return predefined separators
+        cls.mock_mapping_comma_dot = Mock()
+        cls.mock_mapping_comma_dot._get_float_separators.return_value = (",", ".")
+        cls.mock_mapping_dot_comma = Mock()
+        cls.mock_mapping_dot_comma._get_float_separators.return_value = (".", ",")
+        cls.journal = cls.AccountJournal.create(
+            {
+                "name": "Bank",
+                "type": "bank",
+                "code": "BANK",
+                "currency_id": cls.currency_usd.id,
+                "suspense_account_id": cls.suspense_account.id,
+            }
+        )
+        cls.statement_domain = [("journal_id", "=", cls.journal.id)]
+
+    def _get_import_wizard(self, path):
+        return self.AccountStatementImport.with_context(
+            journal_id=self.journal.id, account_statement_import_sheet_file_test=True
+        ).create(
+            {
+                "statement_filename": path,
+                "statement_file": self._data_file(path),
+                "sheet_mapping_id": self.sample_statement_map.id,
+            }
+        )
+
+    def _data_file(self, filename, encoding=None):
+        mode = "rt" if encoding else "rb"
+        with open(path.join(path.dirname(__file__), filename), mode) as file:
+            data = file.read()
+            if encoding:
+                data = data.encode(encoding)
+            return b64encode(data)
+
+    def test_import_csv_file(self):
+        wizard = self._get_import_wizard("fixtures/sample_statement_en.csv")
+        wizard.import_file_button()
+        statement = self.AccountBankStatement.search(self.statement_domain)
+        self.assertEqual(len(statement), 1)
+        self.assertEqual(len(statement.line_ids), 2)
+
+    def test_import_empty_csv_file(self):
+        wizard = self._get_import_wizard("fixtures/empty_statement_en.csv")
+        with self.assertRaises(UserError):
+            wizard.import_file_button()
```

Comment:
It could be useful to use `assertRaisesRegex` instead of `assertRaises` to specify the expected message and be more precise.
```suggestion
        with self.assertRaisesRegex(UserError, expected_msj):
            wizard.import_file_button()
```

Reply:
Although it can prevent to catch a non-desired error, the scope of this call is very limited, and I don't think it's very useful to restrict to a specific error message. On contrary, putting that message and if later the message is changed, refactored to use a general one, etc, the test won't be resilient and you may not know exactly where the problem comes (you get that the raise is not triggered, but it is, but with another message).

Thus, I prefer in general to not do this.